site/content/blog/moving-existing-resources-to-a-new-app.md

```diff
@@ -0,0 +1,113 @@
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+  annotations:
+    kapp.k14s.io/disable-default-label-scoping-rules: ""
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+```
+
+The annotation doesn't affect the app's ability to track such resources, the only consequence of disabling label scoping is that other resources in other apps cannot have the same label selector. For example, with the above change in place, any other app cannot have a deployment with the same label selector, i.e. `app: nginx`.
+
+### Option 2
+Another way to overcome this issue is to use `fallback-on-replace` as an [update strategy](/kapp/docs/latest/apply/#kappk14sioupdate-strategy), which will ask kapp to delete+create the resource on encountering an error while updating it. The update strategy can be used via adding the annotation `"kapp.k14s.io/update-strategy": "fallback-on-replace"`.
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-deployment
+  labels:
+    app: nginx
+  annotations:
+    kapp.k14s.io/update-strategy: "fallback-on-replace"
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        ports:
+        - containerPort: 80
+```
+
+The consequence of using this update-strategy is having downtime as the deployment will be deleted and re-created.
+
+## Moving resources across kapp apps
+
+When we try to create an app with resources that are already part of another app, kapp will through an ownership error that says that the resource is already part of another app.
+```bash
+kapp: Error: Ownership errors:
+- Resource 'deployment/nginx-deployment (apps/v1) namespace: default' is already associated with a different app 'foo' namespace: default (label 'kapp.k14s.io/app=1646214636675344000')
+```
+
+To overcome this, we can use the flag `--dangerous-override-ownership-of-existing-resources` to override the ownership label on those resources. This would remove these resources from the other app and make them part of the current app.
+
+For some resources like Deployment, if label scoping was turned on (default=yes), then its `spec.selector` would also include app label, which would have to be rewritten by kapp when ownership changes. This will cause an immutable field error during update change since k8s deployment does not support changing selector labels.
+
+Even if we disable the label scoping for these resources, kapp would still try to remove the existing label selector and the deployment would still fail as removing the label selector still means that an update to the field is required.
```

Comment:
If we try to manually add the generated label to the `spec.selector` so that kapp doesn't try to make a change to that field, we would get another `Invalid` error as `spec.selector` should have the same labels as that of `spec.template.metadata`, and even if we disable label scoping, kapp would still try to update the `spec.template.metadata` field.
Any thoughts on how we can overcome this? (I couldn't think of anything simple that would work here)
cc @cppforlife